test/external/kickstart_syntax_test.rb

```diff
@@ -0,0 +1,23 @@
+require "test_helper"
+require "English"
+
+class KickstartSyntaxTest < ActiveSupport::TestCase
+  # Kickstart snapshots are generated only for EL7
+  ["RHEL7"].each do |version|
+    Dir.glob('test/unit/foreman/renderer/snapshots/ProvisioningTemplate/provision/*Kickstart*').each do |file|
+      test "#{version} for #{file}" do
+        ksvalidator(file, version)
+      end
+    end
+  end
```

Comment:
I think you can use `describe` as well:
```suggestion
  ["RHEL7"].each do |version|
    describe version do
      Dir.glob('test/unit/foreman/renderer/snapshots/ProvisioningTemplate/provision/*Kickstart*').each do |file|
        test file do
          ksvalidator(file, version)
        end
      end
    end
  end
```

This may show up cleaner in the reporting but not 100% sure. I have more experience with rspec than minitest.

Reply:
Yeah its called `context` in minitest, I can do that sure.